content/posts/efficient-navigation-in-vim.md

```diff
@@ -0,0 +1,140 @@
+
+The alternative is, as described in the next section, to use the `h`, `j`, `k` and `l` keys on the keyboard which are conveniently placed at the center and where the user most likely has their fingers most of the time.
+
+## Using h,j,k,l
+
+**TL;DR: Use for small navigations. Use in combination with relative line numbers.**
+
+`h`, `j`, `k` and `l` are the basic movement keys in Vim. They should be used instead of the usual arrow keys on the keyboard to, as discussed above, keep one's fingers on the home row as much as possible. It takes a little practice but will pay off in the long run.
+
+One hugely important thing to do with these keys is to not press them multiple times in sequence or even hold them down to move several columns or rows. As with many commands in Vim, they can be prefixed with numbers to move multiple times. For example, `12j` can be used rather than pressing the `j` key twelve times which is, obviously, much more efficient (three vs. twelve keystrokes). However, it is recommendable to only use these keys for small movements and mostly to move lines up or down because there are more efficient ways of moving longer vertical distances or horizontal distances in general.
+
+It can be very useful to enable relative line numbers to see at a glance what number to prefix `j` or `k` with to move to a certain line. They can be enabled by adding the following two lines to the configuration file:
+
+```vim
+" Enable relative line numbers
+set relativenumber
+set number
+```
+
+This will always show the absolute number of the line the cursor is currently on and a relative number for all others.
+
+## Navigation Within a Line
+
+**TL;DR: Mostly use `f`. Also consider `w`,`b`,`e`,`^` and `$`.**
+
+So far, we've mostly looked at navigating from one line to another. The next step is to navigate within a line. In many cases, the straight forward thing to do is to use `w` or `W` to move to the next word, `b` or `B` to move to the previous word or even `e` or `E` to move directly to the end of the next word. For all of them, the lowercase variant considers a "word" to be what we intuitively see as one using delimiters like `-`, `/` or `.` to separate one word from another. The uppercase variant considers anything a word that is delimited by whitespace. Obviously, any of these commands can be prefixed with a number to jump multiple words in one go (e.g. `7w`). Other useful commands are `^` which moves to the first non-whitespace character of the line and `$` to move to the last character of a line. Considering the following line
+
+```txt
+abc def-ghi-jkl mno
+```
+
+with the cursor currently on the `g` character, to get to a specific target character pressing the following commands would be the most efficient way to do so:
```

Comment:
I would re-order this sentence `on the g character, pressing the following commands would be the most efficient way to get to a specific target character`